lib/iris/quickplot.py

```diff
@@ -46,6 +47,13 @@ def _title(cube_or_coord, with_units):
 
             if _use_symbol(units):
                 units = units.symbol
+            if units.is_time_reference():
+                # iris.plot uses matplotlib.dates.date2num, which is fixed to the below unit.
+                if mpl_version >= "3.3":
+                    days_since = "1970-01-01"
+                else:
+                    days_since = "0001-01-01"
+                units = "days since {}".format(days_since)
```

Comment:
The change in dates with 3.3 is due to https://github.com/matplotlib/matplotlib/pull/15008.